bgpd/bgp_updgrp_adv.c

```diff
@@ -755,37 +755,34 @@ void subgroup_default_originate(struct update_subgroup *subgrp, int withdraw)
 	}
 
 	if (peer->default_rmap[afi][safi].name) {
+		struct attr attr_tmp = attr;
+		struct bgp_path_info bpi_rmap = {0};
+
+		bpi_rmap.peer = bgp->peer_self;
+		bpi_rmap.attr = &attr_tmp;
+
 		SET_FLAG(bgp->peer_self->rmap_type, PEER_RMAP_TYPE_DEFAULT);
 		for (rn = bgp_table_top(bgp->rib[afi][safi]); rn;
 		     rn = bgp_route_next(rn)) {
 			for (ri = bgp_node_get_bgp_path_info(rn);
 			     ri; ri = ri->next) {
-				struct attr dummy_attr;
-
-				/* Provide dummy so the route-map can't modify
-				 * the attributes */
-				bgp_attr_dup(&dummy_attr, ri->attr);
-				tmp_info.peer = ri->peer;
-				tmp_info.attr = &dummy_attr;
-
 				ret = route_map_apply(
 					peer->default_rmap[afi][safi].map,
-					&rn->p, RMAP_BGP, &tmp_info);
```

Comment:
Not sure if peer structure is being used internally for all cases in route_map_apply. if we are using for any case  then this fix might break that assumptions.